### PR TITLE
[DUOS-2393][risk=no] Remove DataRequestVoteResource

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadworkbench.atlassian.net/browse/DUOS-2393

Removes the `DataRequestVoteResource` and all dependent code. Also cleans up some previously unused code missed in other cleanup tasks. As fallout, there are a number of emails that we have not been sending out and are removed:
* Cancel DAR Request Message
* Collect Message (i.e. `Review case id "DAR-ID" is ready for vote collection.`)
* Flagged Admin Approved DAR Message
* Delegate Responsibilities Message
* Data Custodian Election Approval Message

Also fixes a flaky `DatasetServiceDAOTest` test

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/test/java/org/broadinstitute/consent/http/db/DataAccessRequestDAOTest.java [hunk 1]
```diff
@@ -605,82 +603,82 @@ public void testFindAllDataAccessRequestsForInstitutionArchived() {
...
-        List<DatasetProperty> list = new ArrayList<>();
-        DatasetProperty dsp = new DatasetProperty();
-        dsp.setDataSetId(datasetId);
-        dsp.setPropertyKey(1);
-        dsp.setPropertyValue("Test_PropertyValue");
-        dsp.setCreateDate(new Date());
-        list.add(dsp);
-        datasetDAO.insertDatasetProperties(list);
-    }
+//    @Test
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/consent/http/db/DataAccessRequestDAOTest.java]: nit: remove this commented out code

File: src/test/java/org/broadinstitute/consent/http/db/DatasetDAOTest.java [hunk 2]
```diff
@@ -130,24 +130,24 @@ public void testFindDatasetsByAlias() {
...
-        List<Dataset> datasets = datasetDAO.findNeedsApprovalDatasetByDatasetId(List.of(dataset.getDataSetId()));
-        assertFalse(datasets.isEmpty());
-        assertEquals(1, datasets.size());
-        assertEquals(dac.getDacId(), datasets.get(0).getDacId());
-        assertEquals(consent.getConsentId(), datasets.get(0).getConsentId());
-        assertEquals(consent.getTranslatedUseRestriction(), datasets.get(0).getTranslatedUseRestriction());
-        assertFalse(datasets.get(0).getProperties().isEmpty());
-        assertTrue(datasets.get(0).getNeedsApproval());
-    }
+//    @Test
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/consent/http/db/DatasetDAOTest.java]: nit: remove this commented out code

File: src/test/java/org/broadinstitute/consent/http/db/VoteDAOTest.java [hunk 3]
```diff
@@ -231,18 +179,18 @@ public void testInsertVote() {
...
-        Consent consent = createConsent();
-        Dataset dataset = createDataset();
-        Election election = createDataAccessElection(consent.getConsentId(), dataset.getDataSetId());
-        Vote vote = createDacVote(user.getUserId(), election.getElectionId());
-
-        voteDAO.deleteVoteById(vote.getVoteId());
-        Vote foundVote = voteDAO.findVoteById(vote.getVoteId());
-        assertNull(foundVote);
-    }
+//    @Test
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/consent/http/db/VoteDAOTest.java]: nit: remove this commented out code

File: src/test/java/org/broadinstitute/consent/http/service/DatasetServiceTest.java [hunk 4]
```diff
@@ -219,16 +207,16 @@ public void testUpdateNeedsReviewDataSets(){
...
-    public void testFindNeedsApprovalDataSetsByObjectId() {
-        when(datasetDAO.findNeedsApprovalDatasetByDatasetId(Collections.singletonList(1)))
-                .thenReturn(getDatasets());
-        initService();
-
-        List<Dataset> dataSets = datasetService.findNeedsApprovalDataSetByObjectId(Collections.singletonList(1));
-        assertNotNull(dataSets);
-        assertEquals(dataSets.stream().findFirst().orElseThrow().getDataSetId(), getDatasets().stream().findFirst().orElseThrow().getDataSetId());
-    }
+//    @Test
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/consent/http/service/DatasetServiceTest.java]: nit: remove this commented out code

File: src/test/java/org/broadinstitute/consent/http/service/ElectionServiceTest.java [hunk 5]
```diff
@@ -194,16 +179,16 @@ private void bunchOfDoNothings() throws Exception {
         doNothing().when(consentDAO).updateConsentSortDate(any(), any());
         doNothing().when(dataAccessRequestDAO).updateDataByReferenceId(any(), any(), any(), any(), any(), any());
         doNothing().when(electionDAO).insertAccessAndConsentElection(any(), any());
-        doNothing().when(consentDAO).updateConsentUpdateStatus(any(), any());
+//        doNothing().when(consentDAO).updateConsentUpdateStatus(any(), any());
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/consent/http/service/ElectionServiceTest.java]: nit: remove this commented out code

File: src/test/java/org/broadinstitute/consent/http/service/ElectionServiceTest.java [hunk 6]
```diff
@@ -194,16 +179,16 @@ private void bunchOfDoNothings() throws Exception {
...
         doNothing().when(electionDAO).insertAccessRP(any(), any());
         doNothing().when(electionDAO).updateElectionById(any(), any(), any());
         doNothing().when(electionDAO).updateElectionById(any(), any(), any(), any());
         doNothing().when(electionDAO).archiveElectionById(any(), any());
         doNothing().when(electionDAO).updateElectionStatus(any(), any());
         doNothing().when(emailService).sendResearcherDarApproved(any(), any(), any(), any());
-        doNothing().when(emailService).sendDataCustodianApprovalMessage(any(), any(), any(), any(), any());
+//        doNothing().when(emailService).sendDataCustodianApprovalMessage(any(), any(), any(), any(), any());
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/consent/http/service/ElectionServiceTest.java]: nit: remove this commented out code

File: src/test/java/org/broadinstitute/consent/http/service/ElectionServiceTest.java [hunk 7]
```diff
@@ -194,16 +179,16 @@ private void bunchOfDoNothings() throws Exception {
...
         doNothing().when(emailService).sendClosedDataSetElectionsMessage(any());
-        doNothing().when(voteDAO).deleteVoteById(any());
+//        doNothing().when(voteDAO).deleteVoteById(any());
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/consent/http/service/ElectionServiceTest.java]: nit: remove this commented out code

File: src/test/java/org/broadinstitute/consent/http/service/ElectionServiceTest.java [hunk 8]
```diff
@@ -349,36 +329,36 @@ public void testUpdateElectionByIdWithArchival_Open() {
...
-        try{
-            Election election = service.submitFinalAccessVoteDataRequestElection(sampleElection1.getElectionId(), false);
-            assertNotNull(election);
-            assertEquals(sampleElection1.getElectionId(), election.getElectionId());
-        //function throws exception, need to have a catch block to handle it
-        } catch(Exception e) {
-            Assert.fail("Vote should not have failed");
-        }
-    }
+//    @Test
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/consent/http/service/ElectionServiceTest.java]: nit: remove this commented out code

File: src/test/java/org/broadinstitute/consent/http/service/ElectionServiceTest.java [hunk 9]
```diff
@@ -513,22 +493,22 @@ public void testCheckDataOwnerToCloseElection() {
...
-        when(electionDAO.insertElection(any(), any(), any(), any(), any()))
-                .thenReturn(sampleDatasetElection.getElectionId());
-        initService();
-        List<Election> elections = service.createDataSetElections(sampleElection1.getReferenceId(), Map.of(sampleUserMember,
-            List.of(sampleDataset1)));
-        assertNotNull(elections);
-        assertEquals(1, elections.size());
-        assertEquals(sampleDatasetElection.getReferenceId(), elections.get(0).getReferenceId());
-    }
+//    @Test
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/consent/http/service/ElectionServiceTest.java]: nit: remove this commented out code

File: src/test/java/org/broadinstitute/consent/http/service/VoteServiceTest.java [hunk 10]
```diff
@@ -84,8 +83,8 @@ public void setUp() {
 
     private void doNothings() {
         doNothing().when(voteDAO).updateVote(anyBoolean(), anyString(), any(), anyInt(), anyBoolean(), anyInt(), any(), anyBoolean());
-        doNothing().when(voteDAO).deleteVoteById(anyInt());
-        doNothing().when(voteDAO).deleteVotesByReferenceId(anyString());
+//        doNothing().when(voteDAO).deleteVoteById(anyInt());
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/consent/http/service/VoteServiceTest.java]: nit: remove this commented out code

File: src/test/java/org/broadinstitute/consent/http/service/VoteServiceTest.java [hunk 11]
```diff
@@ -122,15 +121,15 @@ public void testUpdateVote() {
...
-    @Test
-    public void testUpdateVoteById() {
-        Vote v = setUpTestVote(false, false);
-        initService();
-
-        Vote vote = service.updateVoteById(v, v.getVoteId());
-        assertNotNull(vote);
-        assertEquals(v.getVoteId(), vote.getVoteId());
-    }
+//    @Test
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/consent/http/service/VoteServiceTest.java]: nit: remove this commented out code

File: src/test/java/org/broadinstitute/consent/http/service/VoteServiceTest.java [hunk 12]
```diff
@@ -297,100 +296,100 @@ public void testCreateDataOwnersReviewVotes() {
...
-        Vote v = setUpTestVote(false, false);
-        when(voteDAO.findVotesByReferenceIdTypeAndUser("test", 1, VoteType.DATA_OWNER.getValue()))
-                .thenReturn(v);
-        initService();
-
-        Vote vote = service.describeDataOwnerVote("test", 1);
-        assertNotNull(vote);
-        assertEquals(v.getVoteId(), vote.getVoteId());
-    }
+//    @Test(expected = NotFoundException.class)
```
Comment (fboulnois) [inline src/test/java/org/broadinstitute/consent/http/service/VoteServiceTest.java]: nit: remove this commented out code

**Review by fboulnois — APPROVED** [review]
lgtm, mainly suggestions around removing commented out code:

**Comment by rushtong** [comment]
> lgtm, mainly suggestions around removing commented out code:

Thank you - I had intended to do that!

**Review by connorlbark — APPROVED** [review]
👍  contingent on florian's suggestion. olympian level deletes! a new record!

**Pull request merged**